articles/nodejs-structural-comparisons/index.md

```diff
@@ -0,0 +1,562 @@
+
+For example, without a `global` function to define a variable, the variable will be named `undefined`.  
+
+Take a look at the example below.  
+
+```JavaScript
+var name = 'section.io';
+```  
+
+**Output:**  
+```JavaScript
+undefined
+```  
+
+When adding `global` before the variable name, the value becomes defined. This can be applied to many data types and node.js applications.  
+
+The following are examples.  
+
+```JavaScript
+global.name = 'section.io';
+global.number = [0,-10,3.14,-3.14];
+global.alphanum = [name, number];
+```  
+
+**Output:**  
+```JavaScript
+'section.io'
+[ 0, -10, 3.14, -3.14 ]
+[ 'section.io', [ 0, -10, 3.14, -3.14 ] ]
+```  
+
+The picture below is a Node.js window.  
+[Defining Variables](/engineering-education/nodejs-structural-comparisons/define.jpg)  
+_Screen capture_  
+
+Handling different data types inside an individual variable might need a specific structure; square brackets may surround values.  
+
+##### R-Programming
+
+When defining variables in R-Programming, the variable name and an arrow-type symbol (`<-`). For example, the codes below will show variables corresponding to a data type.  
+
+Numerical list:   
+
+```Bash
+numbers <- list(5,7,2,8,0,3,-5)
+# Lists are written in this format.
+numbers
+```  
+
+**Output:**  
+
+1 | 2 | 3 | 4 | 5 | 6 | 7
+--- | --- | --- | --- | --- | --- | ---
+[1] 5 | [1] 7 | [1] 2 | [1] 8 | [1] 0 | [1] 3 | [1] -5
+
+Text and strings:  
+
+```Bash
+text <- c("section.io")
+text
+```  
+
+**Output:**  
+```Bash
+[1] "section.io"
+```  
+
+Numerical lists and matrices:  
+
+```Bash
+x <- list(2,3,4,5,6,7)
+matrix(x,ncol=3)
+# Number of rows or columns is a divisible number.
+```  
+
+To avoid warning messages, choose a number divisible by 6 (the number count in the list). If not, the matrix will either repeat or cut the sequence in the given list.  
+
+**Output:**  
+
+index | [,1] | [,2] | [,3]
+--- | --- | --- | ---
+[1,] | 2 | 4 | 6
+[2,] | 3 | 5 | 7
+
+```Bash
+matrix(x,ncol=9) # More than 6.
+matrix(x,ncol=4) # Less than 6.
+# Both numbers of rows or columns are not divisible by 6.
+```  
+
+**Outputs:**  
+
+More than the number count in the list.  
+index | [,1] | [,2] | [,3] | [,4] | [,5] | [,6] | [,7] | [,8] | [,9]
+--- | --- | --- | --- | --- | --- | --- | --- | --- | ---
+[1,] | 2 | 3 | 4 | 5 | 6 | 7 | 2 | 3 | 4
+
+Less than the number count in the list.  
+ index | [,1] | [,2] | [,3] | [,4]
+ --- | --- | --- | --- | ---
+[1,] | 2  |  4  |  6  |  2
+[2,] | 3  |  5  |  7  |  3
+
+
+##### C/C++
+
+C++ is structured in a different way where compiling a script is required. Although, initializing variables remains similar to other computer languages.  
+
+```C
+int i = 8; //Numerical integers.
+string v = "Section.io"; //Alpha text string.
+```  
+
+While a common practice of declaring or initializing variables is similar to many computer languages, classes could appear different.  
+
+Global and local variables are written typically, as shown below.  
```

Comment:
Don't use typically usually for statements that classify as facts. Writing code is fact based, that is, there is syntax involved. Therefore, avoid using usually, typically. Change throughout the article.

Reply:
The use of 'typical' and other similar words were chosen to consider the concept of depreciation and the future outlook that perhaps at one point it will become unsupported. With that said, I will try to avoid those words as each reviewer has their own expectations.